### PR TITLE
[BPK-2710] Add rating component

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs here https://github.com/Skyscanner/backpack-docs/pull/114 

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/Rating/Classes/BPKRatingBubble.m [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        [self setUp];
+    }
+
+    return self;
+}
+
+- (void)setUp {
+    _lowRatingColor = BPKColor.red500;
+    _mediumRatingColor = BPKColor.yellow500;
+    _highRatingColor = BPKColor.green500;
```
Comment (imchristian) [inline Backpack/Rating/Classes/BPKRatingBubble.m]: Are we using ivars here form some particular reason?

File: Backpack/Rating/Classes/BPKRatingBubble.m [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+- (void)setUpConstraints {
+    CGFloat ratingBubbleVerticalSpacing = [self ratingBubbleVerticalSpacing];
+
+    self.ratingBubbleLabel.translatesAutoresizingMaskIntoConstraints = NO;
+
+    [NSLayoutConstraint activateConstraints:@[
+        [self.heightAnchor constraintEqualToAnchor:self.ratingBubbleLabel.heightAnchor
+                                          constant:2 * ratingBubbleVerticalSpacing],
+        [self.heightAnchor constraintEqualToAnchor:self.widthAnchor],
```
Comment (imchristian) [inline Backpack/Rating/Classes/BPKRatingBubble.m]: Why do we have to constraints here for the same anchor?

File: Backpack/Rating/Classes/BPKRatingBubble.m (hunk 1 above)
Comment (georgegillams) [reply Backpack/Rating/Classes/BPKRatingBubble.m]: Woops!

File: Backpack/Rating/Classes/BPKRating.m [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    CGSize ratingBubbleSize = [self.ratingBubble systemLayoutSizeFittingSize:targetSize];
+    CGSize titleLabelSize = [self.titleLabel systemLayoutSizeFittingSize:targetSize];
+
+    CGFloat height = MIN(MAX(ratingBubbleSize.height, titleLabelSize.height), targetSize.height);
+    CGFloat width = MIN(ratingBubbleSize.width + BPKSpacingMd + titleLabelSize.width, targetSize.width);
+
+    return CGSizeMake(width, height);
+}
+
+#pragma mark - Updates
```
[the remark this answers is not in the document]
Comment (georgegillams) [reply Backpack/Rating/Classes/BPKRating.m]: Well we also need to apply any themed colours that come in, so think it makes sense to have the method for setting the colour.
Can definitely in-line the label setting though

File: Backpack/Rating/Classes/BPKRatingBubble.m (hunk 2 above)
Comment (georgegillams) [reply Backpack/Rating/Classes/BPKRatingBubble.m]: Cos i wrote it wrong 😆

**Pull request merged**